core/trino-main/src/main/java/io/trino/sql/planner/plan/TableScanNode.java

```diff
@@ -75,13 +76,26 @@ public static TableScanNode newInstance(
         return new TableScanNode(id, table, outputs, assignments, TupleDomain.all(), Optional.empty(), updateTarget, useConnectorNodePartitioning);
     }
 
-    /*
-     * This constructor is for JSON deserialization only. Do not use.
-     * It's marked as @Deprecated to help avoid usage, and not because we plan to remove it.
-     */
-    /* TODO @DoNotCall once it's applicable to constructors */
+    @DoNotCall // For JSON serialization only
     @JsonCreator
-    public TableScanNode(
+    public static TableScanNode fromJson(
+            @JsonProperty("id") PlanNodeId id,
```

Comment:
Do we still need to annotate parameters, if the parameter names match the JSON? I know that records are handled automatically.